### DISCUSSION TITLE
CISCO ASA FILEBEAT MODULE

### DISCUSSION THREAD

**Discussion by niiampim** [Security-Onion-Solutions/securityonion#9102] [Unsupported Versions]
Hello,
I am attempting to use cisco filebeat module to ingest logs into SO.
I have created
1. Hostgroups and Portgroups
2. configured /opt/so/saltstack/local/pillars/minion/myfile.sls with the cisco module configuration.
3. Applied state for firewall and filebeat and can confirm via iptables that the manager is listening on port 9001

However, filebeat docker container does not seem to run.

docker log so-filebeat gives below:
Exiting: Failed to start crawler: creating module reloader failed: error checking input configuration: No paths were defined for input accessing config

Anyone with any experience please assist

**Comment by dougburks** [comment]
> Exiting: Failed to start crawler: creating module reloader failed: error checking input configuration: No paths were defined for input accessing config

This error message seems to indicate that there is a problem with your Filebeat configuration. Please double-check your configuration:
https://docs.securityonion.net/en/2.3/filebeat.html
https://www.elastic.co/guide/en/beats/filebeat/8.4/filebeat-module-cisco.html

**Reply by niiampim** [reply]
I am not sure where to look in the configuration file as i haven't made any such changes. 
Also, i noticed there are 2 filebeat folders;
/opt/conf and /saltstack/local

which takes precedence and which file should i be looking to change any configuration

**Reply by niiampim** [reply]
Update:
I got the filebeat container to work. I changed the path to "/opt/so/conf/filebeat/etc/module-setup.yml". Is this best practice? I still did not get the logs from the cisco asa

this is the filebeat.yml in /opt/so/saltstack/local/salt/filebeat/etc

`#==========================  Modules configuration ============================
filebeat.config.modules:
  enabled: true
 #path: ${path.config}/modules.d/*.yml
  path: "/opt/so/conf/filebeat/etc/module-setup.yml"
`

**Reply by niiampim** [reply]
Update:
I am receiving logs and can view them in Hunt in SO.  Below is my filebeat.yml file. I want to find out if this is the best practice of adding the cisco ASA config under filebeat.input.
I had already configured similar in /opt/so/saltstack/local/pillars/minion/myfile.sls. 

Unfortunately i am not seeing this in Kibana.

`
#==========================  Modules configuration ============================
filebeat.config.modules:
  enabled: true
 #path: ${path.config}/modules.d/*.yml
  path: "/opt/so/conf/filebeat/etc/module-setup.yml"

filebeat.modules:
  enabled: true
#=========================== Filebeat prospectors =============================

# List of prospectors to fetch data.
filebeat.inputs:
#------------------------------ Log prospector --------------------------------
- type: udp
  enabled: true
  host: "0.0.0.0:514"
  fields:
    module: syslog
    dataset: syslog
  pipeline: "syslog"
  index: "so-syslog"
  processors:
  - drop_fields:
      fields: ["source", "prospector", "input", "offset", "beat"]
  fields_under_root: true

- type: tcp
  enabled: true
  host: "0.0.0.0:514"
  fields:
    module: syslog
    dataset: syslog
  pipeline: "syslog"
  index: "so-syslog"
  processors:
  - drop_fields:
      fields: ["source", "prospector", "input", "offset", "beat"]
  fields_under_root: true

- type: udp
  enabled: true
  host: "0.0.0.0:9001"
  fields:
    module: cisco
    dataset: cisco.asa
  pipeline: "filebeat-7.13.4-cisco-asa-asa-ftd-pipeline"
`

**Reply by weslambert** [reply]
Assuming you are using a supported module, you should not have to make any changes to `filbeat.yml`, or any files other than the associated pillar file.

You should be able trace the events through the pipeline using the methods below.

# Check Logstash pipeline stats
`so-logstash-pipeline-stats $pipeline`

# Check if Elasticsearch pipelines are loaded
`so-elasticsearch-pipelines-list`

# Check Elasticseach pipeline stats
`so-elasticsearch-pipeline-stats $pipeline`

Thanks,
Wes
